### PR TITLE
feat: add new method for CommandClient

### LINKED ISSUE #713: Command Client missing capability to passes additional query parameters for Get command
IssueGetCommandByName() doesn't have a parameter for addition query parameters.

Need to add new method `IssueGetCommandByNameWithQueryParms` to allows the user query parameters to be added to the Get command request.

### PR DESCRIPTION
**Pull request by hahattan**
add 'IssueGetCommandByNameWithQueryParams' method to allow the user
query parameters to be added to the Get command request.

fixes #713

Signed-off-by: Chris Hung <chris@iotechsys.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

1. use `CommandClient` from this branch and verify the new method is working

### CODE REVIEW

File: clients/http/command_test.go
Comment (lenny-intel) [inline clients/http/command_test.go:61]: Can you add code to verify the query params where actually in the resulting REST request.

File: clients/http/command.go
Comment (lenny-intel) [inline clients/http/command.go:77]: Is it ok that the `dsPushEvent`  and `dsReturnEvent` parameters are not sent. I.e. does the Device Service have a default for these if not present? If not, we need to add defaults here is they are not present in the passed in `queryParams`

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: clients/http/command_test.go
Comment (cloudxxx8) [reply clients/http/command_test.go:61]: @hahattan  we have the default values in the core-command, so we should just add the mock function to behave as the core-command default values here

File: clients/http/command.go
Comment (cloudxxx8) [reply clients/http/command.go:77]: @lenny-intel since we have the default value in core-command, I don't think we should add the default value in the client library to prevent the double maintenance effort if we want to change the default values in the future.

Comment (hahattan) [reply clients/http/command.go:77]: Device service also have default value:
https://docs.edgexfoundry.org/2.0/design/adr/device-service/0011-DeviceService-Rest-API/#query-parameters

File: clients/http/command_test.go
Comment (lenny-intel) [reply clients/http/command_test.go:75]: Need to get this merged ASAP, once TSC approved after code Freeze

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**
